scorers.py

```diff
@@ -39,10 +39,34 @@ def score_corpus(self, ref, out):
       A tuple containing a single value for the BLEU score and a string summarizing auxiliary information
     """
     if self.case_insensitive:
-      bleu = nltk.translate.bleu_score.corpus_bleu([[corpus_utils.lower(x)] for x in ref], corpus_utils.lower(out), weights=self.weights)
-    else:
-      bleu = nltk.translate.bleu_score.corpus_bleu([[x] for x in ref], out, weights=self.weights)
-    return bleu, None
+      ref = corpus_utils.lower(ref)
+      out = corpus_utils.lower(out)
+
+    num_prec = Counter()
```

Comment:
This code is copy-pasted from another part of the file, which also isn't good software engineering practice, as it can lead to you modifying one part of the code but not the other. For this function, I'd calculate the statistics, then call `calc_cached_score()` directly.